### PR TITLE
add link to swap between blocked and all client queries

### PR DESCRIPTION
**Pull request by stuXORstu**
Signed-off-by: stuXORstu <stuxorstu@gmail.com>

**By submitting this pull request, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

`Inserts a quick handy link to swap between blocked and all queries for clients.  Seemed like a handy thing to have since when troubleshooting potential blocks I end up first looking at blocked queries, and then if I can't spot anything want to look at all queries to confirm if recent activity is not being blocked - so having a quick link to jump between those two states seemed useful to me, particularly when using a mobile device and not wanting to mess about with multiple tabs`

**How does this PR accomplish the above?:**

`Modification of the existing code that shows queries types and provides context to the user on the queries.php page`

**What documentation changes (if any) are needed to support this PR?:**

`None - it is fairly straightforward (hopefully!!)`

### CODE REVIEW

**Review by PromoFaux — DISMISSED** [review]
I've tested and it does exactly what it says on the tin. I like the idea, but I'd like a second opinion from @pi-hole/web-maintainers first, in case I'm missing a reason why not to include this...

**Review by yubiuser — CHANGES_REQUESTED** [review]
Thanks for your PR.

I had spend some time to simplify the logic here. Most importantly: we have only a single place where we use `type` at all - it's the dashboard link in the "Top Clients (blocked) only)". The request will only work, **if also a client is submitted**
https://github.com/pi-hole/AdminLTE/blob/b57ff6aeec72e99f0226e339fcaf55b3567ee6ae/api_FTL.php#L297-L300

Therefore I think the logic can be a bit simplified: 
1) don't add `if(isset($_GET["type"]) && $_GET["type"] === "blocked")` to `if($setupVars["API_QUERY_LOG_SHOW"] === "all")`
2) remove line 33-35
3) add `blocked` in `$showing .= " queries for client ".htmlentities($_GET["client"]);`

**Comment by stuXORstu** [comment]
Thanks for your advice and clarifying the logic too.  I had initially added some cautious additional logic because I was not sure whether there might have been some ways to exercise params that I could not see from my simple pi-hole setup.  If there is really no point to that as you say, then I've removed the additional complexity as you suggest.

**Comment by yubiuser** [comment]
I didn't meant to be so permissive ;-)
See my comments. Otherwise PR looks good.

File: queries.php
Comment (yubiuser) [inline queries.php]:
```suggestion
	if(isset($_GET["type"]) && $_GET["type"] === "blocked")
```

File: queries.php
Comment (yubiuser) [inline queries.php]:
```suggestion

```

File: queries.php
Comment (yubiuser) [inline queries.php]:
```suggestion
// Add switch between showing all queries and blocked only
```

File: queries.php
Comment (yubiuser) [inline queries.php]:
```suggestion
	if (isset($_GET["type"]) && $_GET["type"] === "blocked") {
		$showing .= "\">show all</a>";
	} else {
		$showing .= "&type=blocked\">show blocked only</a>";
	}
```

**Review by yubiuser — CHANGES_REQUESTED** [review]
Bildschirmfoto zu 2022-03-10 21-01-39
Bildschirmfoto zu 2022-03-10 21-01-45

**Comment by rdwebdesign** [comment]
If all requested changes are applied, the code will end up with 2 `if` statements with the same expression:

```php
    if (isset($_GET["type"]) && $_GET["type"] === "blocked")
    {
        $showing .= " blocked";
    }

    // Add switch between showing all queries and blocked only
    $showing .= " queries for client ".htmlentities($_GET["client"]).", <a href=\"?client=".htmlentities($_GET["client"]);

    if (isset($_GET["type"]) && $_GET["type"] === "blocked") {
        $showing .= "\">show all</a>";
    } else {
        $showing .= "&type=blocked\">show blocked only</a>";
    }
```

I think using only one `if` will make the code more readable:

```php
    if (isset($_GET["type"]) && $_GET["type"] === "blocked") {
        // Show Blocked queries for this client + link to all
        $showing .= ' blocked queries for client '.htmlentities($_GET["client"]);
        $showing .= ', <a href="?client='.htmlentities($_GET["client"]).'">show all</a>';
    } else {
        // Show All queries for this client + link to show only blocked
        $showing .= ' all queries for client '.htmlentities($_GET["client"]);
        $showing .= ', <a href="?client='.htmlentities($_GET["client"]).'&type=blocked">show blocked only</a>';
    }
```

**Comment by stuXORstu** [comment]
Yeah that is more readable than trying to fit into the smaller variable string concatenation at a time.  Thanks, I've stitched it all together in another commit.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**
